src/Checkout/Mixin.js

```diff
@@ -628,19 +643,33 @@ export default {
          * Check that the current checkout step is allowed to be accessed.
          * Returns false if the stage is valid.
          */
-        handleInvalidCheckout(checkoutView = Stage.DEFAULT) {
-            if (!this.currentCheckout.stage) return false
+        handleInvalidCheckout(checkoutView = Stage.DEFAULT, uid) {
+            let serverStage = Tell.serverVariable(`serverStage.${uid}`)
+
+            if (!serverStage) serverStage = 1
 
             if (this.currentCheckout.stage === checkoutView) return false
 
             if (this.currentCheckout.stage >= checkoutView &&
-                this.currentCheckout.stage <= Stage.COMPLETE) return false
+                this.currentCheckout.stage < Stage.COMPLETE) return false
+
+            // Validate server stage vs Client stage
+            if (checkoutView <= serverStage &&
+                serverStage < Stage.COMPLETE) return false
 
             /**
              * We set the loading state to true before making a client side redirect here to stop
              * the user from performing further actions.
              */
             this.formIsLoading = true
+
+            /**
+             * The user will have been trying to do something funky to get here,
+             * such as navigating backwards after competing checkout
+             * We should give their cart a new uid to ensure they can't update a completed order
+             */
+            this.activeCartCollection.uid = Tell.randomUid()
```

Comment:
IIRC we want to only do this if the current checkout url UID === activeCart UID so that the cart id doesn't needlessly get changed multiple times